File: src/model/Campanha.js
const { Sequelize } = require('sequelize');
const sequelize = require('../utils/db');
const { Usuario } = require('./Usuarios');

const Campanha = sequelize.define('campanhas', {
  campanha_id: {
    type: Sequelize.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
//   empresa_id: {
//     type: Sequelize.INTEGER,
//     allowNull: false,
//     references: { model: 'usuarios', key: 'usuario_id' },
//   },
  titulo: {
    type: Sequelize.STRING(100),
    allowNull: false,
    validate: { notEmpty: true },
  },
  descricao: {
    type: Sequelize.TEXT,
    allowNull: true,
  },
  imagem_url: {
    type: Sequelize.STRING(255),
    allowNull: true,
  },
  recompensas: {
    type: Sequelize.ARRAY(Sequelize.INTEGER),
    allowNull: true,
    defaultValue: [],
  },
  data_inicio: {
    type: Sequelize.DATE,
    allowNull: false,
  },
  data_fim: {
    type: Sequelize.DATE,
    allowNull: false,
  },
  ativa: {
    type: Sequelize.BOOLEAN,
    defaultValue: true,
  },
  data_cadastro: {
    type: Sequelize.DATE,
    defaultValue: Sequelize.literal('CURRENT_TIMESTAMP'),
  },
  data_update: {
    type: Sequelize.DATE,
    defaultValue: Sequelize.literal('CURRENT_TIMESTAMP'),
  },
}, {
  schema: 'public',
  tableName: 'campanhas',
  timestamps: false,
//   indexes: [
//     { fields: ['empresa_id'] },
//     { fields: ['ativa', 'data_inicio', 'data_fim'] },
//   ]
});

Campanha.belongsTo(Usuario, { foreignKey: 'empresa_id', as: 'empresa' });

module.exports = { Campanha };

File: src/main/java/com/elo/api/models/Campanha.java
package com.elo.api.models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "campanhas", schema = "public")
@Data
@NoArgsConstructor
public class Campanha {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "campanha_id")
    private Integer campanhaId;

    @ManyToOne
    @JoinColumn(name = "empresa_id", nullable = false)
    private Usuario empresa;

    @Column(nullable = false, length = 100)
    private String titulo;

    @Column(columnDefinition = "TEXT")
    private String descricao;

    @Column(name = "imagem_url")
    private String imagemUrl;

    @ElementCollection
    @CollectionTable(name = "campanha_produtos",
            joinColumns = @JoinColumn(name = "campanha_id"))
    @Column(name = "produto_id")
    private List<Integer> produtos = new ArrayList<>();

    @ElementCollection
    @CollectionTable(name = "campanha_recompensas",
            joinColumns = @JoinColumn(name = "campanha_id"))
    @Column(name = "recom_id")
    private List<Integer> recompensas = new ArrayList<>();

    @Column(name = "data_inicio", nullable = false)
    private LocalDateTime dataInicio;

    @Column(name = "data_fim", nullable = false)
    private LocalDateTime dataFim;

    private Boolean ativa = true;

    @Column(name = "data_cadastro")
    private LocalDateTime dataCadastro = LocalDateTime.now();

    @Column(name = "data_update")
    private LocalDateTime dataUpdate;
}
